Gen2: ImageManip rotated/arbitrary cropping, camera controls

alex-luxonis opened the pull request:
Add types defined similar to some OpenCV ones:
- `Point2f: {float x, y;}`
- `Size2f: {float width, height;}`
- `RotatedRect: {Point2f center; Size2f size; float angle;}`

And for arbitrary cropping, I was thinking on defining a sequence of 4 points in clockwise order forming a quadrilateral, that would get mapped to a rectangle, whose size we specify (so resizing + other warp transforms can be applied). First sequence point would get mapped to the top-left corner of the rectangle.
```
 0___1       0 ____ 1
 /   \   ->   |    |
/_____\       |____|
3     2      3      2
```

These are still open to ideas for improvement.

themarpe commented:
Does the intermediate rectangle size adds any additional benefits compared to just the final output size?

alex-luxonis commented:
> Does the intermediate rectangle size adds any additional benefits compared to just the final output size?

For the rotated rectangle (center, size, angle), it describes the crop area, that can be further resized.
For the quadrilateral (that I renamed it now to `warpFourPoints`), I removed the size specification, and it is auto-computed based on the points geometry, if the final resize is not specified.

themarpe commented on include/depthai-shared/datatype/RawCameraControl.hpp:
```diff
@@ -9,16 +9,258 @@
+
+    void setCommand(Command cmd, bool value = true) {
+        uint64_t mask = 1ull << (uint8_t)cmd;
+        if(value)
```
Either inline or with `{}`

alex-luxonis replied:
Wanted to have it inlined initially, but clangformat expanded it: 
https://github.com/luxonis/depthai-shared/pull/16/commits/2ece34cc5531147aa0b0f73f02e853211f17c061#diff-0c285f8d7cb2334c68a2f38b0fafa637d8be3e37cd4c2c072f03e7a2a505c75bL226
Will add `{}`

themarpe replied:
Saw that behaviour as well. Will have to check if I can modify clangformat for this case

themarpe commented on include/depthai-shared/datatype/RawImageManipConfig.hpp:
```diff
@@ -18,21 +18,81 @@ struct RawImageManipConfig : public RawBuffer {
+        bool enableWarpMatrix = false;
+
+        // clockwise
+        float rotationAngle;
```
Could be `rotationAngleRad` or `rotationAngleDeg` and the `ImageManipConfig` wrapper calculates the correct value.
Then `angleInRadians` could be removed.
^ Only applicable if it affects this field only IMO

alex-luxonis replied:
So I wanted to avoid this conversion math in the host library, but yeah, it's simple enough in this case that we can go with it. I think to go with degrees, as that's the case with most OpenCV API also.
